Src/libCZI/CziParse.cpp

```diff
@@ -201,45 +201,49 @@ using namespace libCZI;
 
     // TODO: we can add a couple of consistency checks here
 
-    // now read the AttachmentEntries
-    std::uint64_t attachmentEntriesSize = ((std::uint64_t)attachmentDirSegment.data.EntryCount) * sizeof(AttachmentEntryA1);
-
-    std::unique_ptr<AttachmentEntryA1, decltype(free)*> pBuffer((AttachmentEntryA1*)malloc((size_t)attachmentEntriesSize), free);
-
-    // now read the attachment-entries
-    try
-    {
-        str->Read(offset + sizeof(attachmentDirSegment), pBuffer.get(), attachmentEntriesSize, &bytesRead);
-    }
-    catch (const std::exception&)
-    {
-        std::throw_with_nested(LibCZIIOException("Error reading FileHeaderSegment", offset + sizeof(attachmentDirSegment), attachmentEntriesSize));
-    }
-
-    if (bytesRead != attachmentEntriesSize)
+    // an empty attachment-directory (with zero entries) is valid and in this
+    //  case we can skip the rest of the processing
+    if (attachmentDirSegment.data.EntryCount > 0)
     {
-        CCZIParse::ThrowNotEnoughDataRead(offset + sizeof(attachmentDirSegment), attachmentEntriesSize, bytesRead);
-    }
+        // now read the AttachmentEntries
+        const uint64_t attachmentEntriesSize = static_cast<uint64_t>(attachmentDirSegment.data.EntryCount) * sizeof(AttachmentEntryA1);
+        unique_ptr<AttachmentEntryA1, decltype(free)*> pBuffer(static_cast<AttachmentEntryA1*>(malloc((size_t)attachmentEntriesSize)), free);
 
-    for (int i = 0; i < attachmentDirSegment.data.EntryCount; ++i)
-    {
-        ConvertToHostByteOrder::Convert(pBuffer.get() + i);
+        // now read the attachment-entries
+        try
+        {
+            str->Read(offset + sizeof(attachmentDirSegment), pBuffer.get(), attachmentEntriesSize, &bytesRead);
+        }
+        catch (const std::exception&)
+        {
+            std::throw_with_nested(LibCZIIOException("Error reading FileHeaderSegment", offset + sizeof(attachmentDirSegment), attachmentEntriesSize));
+        }
 
-        const AttachmentEntryA1* pSrc = pBuffer.get() + i;
-        CCziAttachmentsDirectoryBase::AttachmentEntry ae;
-        bool b = CheckAttachmentSchemaType((const char*)&pSrc->SchemaType[0], 2);
-        if (b == false)
+        if (bytesRead != attachmentEntriesSize)
         {
-            // TODO: what do we do if we encounter this...?
-            continue;
+            CCZIParse::ThrowNotEnoughDataRead(offset + sizeof(attachmentDirSegment), attachmentEntriesSize, bytesRead);
```

Comment:
As codecov warns we may wish to add exception tests for these methods w/ curated "malformed" input data in case with mismatched attachmentEntriesSize and actual attachment sizes.

Reply:
Yes, that would be desirable - and the current state of code-coverage still has room for improvement. Overall, we are at 66% currently.